dataflow_transfer/dataflow_transfer.py

```diff
@@ -1,22 +1,55 @@
-import yaml
 import logging
 import os
 
+from dataflow_transfer.run_classes.illumina_runs import (
+    NovaSeqXPlusRun,
+    NextSeqRun,
+    MiSeqRun,
+)
+from dataflow_transfer.run_classes.ont_runs import PromethIONRun, MinIONRun
+from dataflow_transfer.run_classes.element_runs import AvitiRun
+
 logger = logging.getLogger(__name__)
 
 
-def process_run(run_dir, sequencer, config):
-    # For each run dir:
-    # check the status of the run based on the current state of the run dir
-    # If the run is ongoing (i.e. final file does not exist):
-    # start an rsync process in the background to transfer the data to Miarka (with run-one to avoid multiple concurrent transfers)
-    # If the run is complete (i.e. final file exists):
-    # update the "Sequenced" satus to True
-    # initiate a final transfer
-    # update the "transferred" status to True
-    pass
+def get_run_object(run_dir, sequencer, config):
+    if sequencer == "NovaSeqXPlus":
+        return NovaSeqXPlusRun(run_dir, config)
+    elif sequencer == "NextSeq":
+        return NextSeqRun(run_dir, config)
+    elif sequencer == "MiSeq":
+        return MiSeqRun(run_dir, config)
+    elif sequencer == "PromethION":
+        return PromethIONRun(run_dir, config)
+    elif sequencer == "MinION":
+        return MinIONRun(run_dir, config)
+    elif sequencer == "Aviti":
+        return AvitiRun(run_dir, config)
+    else:
+        return None
```

Comment:
I believe the script would end here if one starts it with a specific run folder. I guess that might be on the todo-list though?

Reply:
Simplest solution is to make sequencer type a required argument when running it for a specific run?

Reply:
Ah, yes. That was on the todo-list but I forgot about it, thanks for pointing it out!

Reply:
I added a `get_run_info` function to extract it from the path